input/script_utils.go

```diff
@@ -923,14 +925,28 @@ func CommitSpendNoDelay(signer Signer, signDesc *SignDescriptor,
 	}
 
 	// Finally, we'll manually craft the witness. The witness here is the
-	// exact same as a regular p2wkh witness, but we'll need to ensure that
-	// we use the tweaked public key as the last item in the witness stack
-	// which was originally used to created the pkScript we're spending.
+	// exact same as a regular p2wkh witness, depending on the value of the
+	// tweakless bool.
 	witness := make([][]byte, 2)
 	witness[0] = append(sweepSig, byte(signDesc.HashType))
-	witness[1] = TweakPubKeyWithTweak(
-		signDesc.KeyDesc.PubKey, signDesc.SingleTweak,
-	).SerializeCompressed()
+
+	// TODO(roasbeef): optionally can make this implicit, by checking if
+	// the SingleTweak is populated or not.
```

Comment:
Is it safer? Explicit means the caller _must_ know what type they want. Things were slightly more simplicity before, but then I reversed it as there was a bug that would've been caught if the caller was forced to know what they wanted. It's also usually the case that the sign desc is passed through several call stacks, so it may be difficult to ensure things are properly set vs having the caller know what type of commitment they really want.

Reply:
There are two sides to it. Yes, adding redundant parameters can catch bugs. But it can also camouflage bugs. In this case it could perhaps be that `SingleTweak` is set somewhere and we never notice because the `tweakless` bool makes us ignore it.

I think in a case like this where the pros and cons are not very clear (also signaled by the TODO), I'd probably choose the minimum code / parameters option.

Reply:
The TODO there was more or less just me recording thoughts as I was going, but I'm on the side of having the explicit parameter since an earlier version that didn't resulted in a bug that's now fixed. The TOOD has been removed in the latest diff.